### PR TITLE
Plugin E2E: Improved support for multiple users and roles

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

While working on docs for how to handle auth in plugin-e2e, I realised we offer no way to create users. Being able to create users might not be necessary for most plugins, but if the plugin supports RBAC, the plugin author will need a way to test fine grained access control. 

This PR introduces a new fixture command called `createUser`. This will allow the user of plugin-e2e to create users with certain roles. These users can then be used in subsequent playwright projects to test fine grained access control.

Basic example using the default admin:admin user (RBAC not needed)

```typescript
projects: [
    // Login to Grafana with admin user and store the cookie on disk for use in other tests
    {
      name: 'authenticate',
      testDir: 'node_modules/@grafana/plugin-e2e/dist/auth',
      testMatch: [/.*auth\.setup\.ts/],
    },
    // Run all tests in parallel using user with admin role
    {
      name: 'admin',
      testDir: './tests/admin',
      use: {
        ...devices['Desktop Chrome'],
        storageState: 'playwright/.auth/admin.json',
      },
      dependencies: ['authenticate'],
    }
  ],
```

Advanced example where a new user with `Viewer` role is created. All tests in the `./tests/viewer` folder will be ran with the new user that has the viewer role. 

```typescript
projects: [
    // Create new user with viewer role and store the cookie on disk for use in other tests
    {
      name: 'createUserAndAuthenticate',
      testDir: 'node_modules/@grafana/plugin-e2e/dist/auth',
      testMatch: [/.*auth\.setup\.ts/],
      use: {
        user: {
          user: 'viewer',
          password: 'password',
          role: 'Viewer',
        },
      },
    },
    // Run all tests in parallel using user with viewer role
    {
      name: 'viewer',
      testDir: './tests/viewer',
      use: {
        ...devices['Desktop Chrome'],
        storageState: 'playwright/.auth/viewer.json',
      },
      dependencies: ['createUserAndAuthenticate'],
    },
  ],
```

More details in the tsdocs! 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/plugin-e2e@0.15.0-canary.726.e0673f6.0
  # or 
  yarn add @grafana/plugin-e2e@0.15.0-canary.726.e0673f6.0
  ```
</details>

### CODE REVIEW

File: packages/plugin-e2e/playwright.config.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default defineConfig<PluginOptions>({
   /* Fail the build on CI if you accidentally left test.only in the source code. */
   forbidOnly: !!process.env.CI,
   /* Retry on CI only */
-  retries: process.env.CI ? 2 : 0,
+  retries: process.env.CI ? 1 : 0,
```
Comment (academo) [inline packages/plugin-e2e/playwright.config.ts:16]: Is this change intended?

**Review by academo — APPROVED** [review]

File: packages/plugin-e2e/playwright.config.ts (hunk 1 above)
Comment (sunker) [reply packages/plugin-e2e/playwright.config.ts:16]: oh, no it's wasn't supposed to be committed. thanks!
